pcgc/pcgc.go

```diff
@@ -1,20 +1,153 @@
+// Package pcgc is a service client for Cloud Manager API.
+//
+// To be able to interact with this API, you have to
+// create a new service:
+//
+//     s := pcgc.NewService(nil)
+//
+// The Service struct has all the methods you need
+// to interact with Cloud Manager API.
+//
 package pcgc
 
 import (
+	"bytes"
 	"context"
-	"fmt"
+	"encoding/json"
+	"github.com/bobziuchkovski/digest"
+	"github.com/google/go-querystring/query"
+	"io"
 	"net/http"
+	"reflect"
+	"runtime"
 )
 
-// DoGetRequest creates a new GET HTTP request, executes it, and returns the response
-func DoGetRequest(url string) (*http.Response, error) {
-	ctx, onCancel := context.WithTimeout(context.Background(), RequestTimeout)
-	defer onCancel()
+const (
+	// Version Library version
+	Version = "v1"
+	// DefaultUserAgent User agent
+	DefaultUserAgent = "pcgc/" + Version + " (" + runtime.GOOS + "; " + runtime.GOARCH + ")"
+	// DefaultURL api base url
+	DefaultURL = "http://localhost:8080"
```

Comment:
nit: DefaultURL doesn't make sense since except for development, Ops Manager will be installed remotely - if users are required to pass it, then the baseURL should be required (thus, we can avoid unnecessary optional logic, which may trip users of this library up)

Reply:
yeah I was going to add a setter to change this and then I forgot